sql/src/main/java/io/crate/metadata/doc/DocSchemaInfo.java

```diff
@@ -323,6 +323,17 @@ public void update(ClusterChangedEvent event) {
         return udfFunctions;
     }
 
+    /**
+     * checks if metaData contains a particular index and
+     * invalidates its aliases if so
+     */
+    @VisibleForTesting
+    void invalidateFromIndex(Index index, MetaData metaData) {
+        if (metaData.hasIndex(index.getName())) {
```

Comment:
Could also call `index` directly but add a null check on the result before calling `getAliases`. It would avoid doing the hashing + equals checks twice.